Index: src/fx.js
===================================================================
--- src/fx.js	(revision 5946)
+++ src/fx.js	(working copy)
@@ -205,7 +205,7 @@
 			complete: fn || !fn && easing ||
 				jQuery.isFunction( speed ) && speed,
 			duration: speed,
-			easing: fn && easing || easing && easing.constructor != Function && easing
+			easing: fn && easing || easing && easing.constructor.toString() != Function.toString() && easing
 		};
 
 		opt.duration = jQuery.fx.off ? 0 : typeof opt.duration == 'number' ? opt.duration :
