Index: compat-1.0.js
===================================================================
--- compat-1.0.js	(revision 403)
+++ compat-1.0.js	(working copy)
@@ -32,7 +32,7 @@
   var o = e[i];
     
   // Handle event unbinding
-  jQuery.fn["un"+o] = function(f){ return this.unbind(o, f); };
+  jQuery.fn["un"+o] = jQuery.fn["un"+o] || function(f){ return this.unbind(o, f); };
     
   // Finally, handle events that only fire once
   jQuery.fn["one"+o] = function(f){
