*** jquery-1.4.1.js	Fri Feb  5 10:43:33 2010
--- jquery-141-patched.js	Fri Feb  5 09:17:59 2010
***************
*** 930,935 ****
--- 930,936 ----
  	// http://thinkweb2.com/projects/prototype/detecting-event-support-without-browser-sniffing/
  	var eventSupported = function( eventName ) { 
  		var el = document.createElement("div"); 
+                 el = el.wrappedJSObject || el;
  		eventName = "on" + eventName; 
  
  		var isSupported = (eventName in el); 
