Index: jquery.js
===================================================================
RCS file: /home/cvs/repository/mclass/javaserver/web/site/js/Attic/jquery.js,v
retrieving revision 1.1.2.1
diff -u -r1.1.2.1 jquery.js
--- jquery.js	5 Oct 2007 15:57:50 -0000	1.1.2.1
+++ jquery.js	11 Oct 2007 21:50:53 -0000
@@ -637,7 +637,7 @@
 
 	css: function(e,p) {
 		if ( p == "height" || p == "width" ) {
-			var old = {}, oHeight, oWidth, d = ["Top","Bottom","Right","Left"];
+			var old = {}, oHeight, oWidth, d = ( p == "height" ? ["Top","Bottom"] : ["Right","Left"] );
 
 			jQuery.each( d, function(){
 				old["padding" + this] = 0;
