Index: core.js
===================================================================
--- core.js	(revision 5091)
+++ core.js	(working copy)
@@ -32,6 +32,8 @@
 // Is it a simple selector
 var isSimple = /^.[^:#\[\.]*$/;
 
+var expando = "jQuery" + (new Date()).getTime(), uuid = 0, windowData = {};
+
 jQuery.fn = jQuery.prototype = {
 	init: function( selector, context ) {
 		// Make sure that a selection was provided
@@ -607,8 +609,6 @@
 	return target;
 };
 
-var expando = "jQuery" + (new Date()).getTime(), uuid = 0, windowData = {};
-
 // exclude the following css properties to add px
 var exclude = /z-?index|font-?weight|opacity|zoom|line-?height/i;
 
