Index: ui.droppable.js
===================================================================
--- ui.droppable.js	(revision 4501)
+++ ui.droppable.js	(working copy)
@@ -18,9 +18,9 @@
 		
 		//Prepare the passed options
 		this.options = $.extend({}, options);
-		var o = this.options; var accept = o.accept;
+		var o = this.options; var accept = o.accept; var self = this;
 		$.extend(o, {
-			accept: o.accept && o.accept.constructor == Function ? o.accept : function(d) {
+			accept: o.accept && o.accept.constructor == Function ? function() { return accept.apply(self, arguments); } : function(d) {
 				return $(d).is(accept);	
 			},
 			tolerance: o.tolerance || 'intersect'		
