Index: src/core.js
===================================================================
--- src/core.js	(revision 5819)
+++ src/core.js	(working copy)
@@ -90,9 +90,6 @@
 		return this.length;
 	},
 
-	// The number of elements contained in the matched element set
-	length: 0,
-
 	// Get the Nth element in the matched element set OR
 	// Get the whole matched element set as a clean array
 	get: function( num ) {
