diff --git a/jquery.ui-1.5b4/ui.tabs.js b/jquery.ui-1.5b4/ui.tabs.js
index aefad9a..25c6b65 100644
--- a/jquery.ui-1.5b4/ui.tabs.js
+++ b/jquery.ui-1.5b4/ui.tabs.js
@@ -325,7 +325,7 @@
 				index = this.$tabs.length; // append by default
 
 			var o = this.options;
-			var $li = $(o.tabTemplate.replace(/#\{href\}/, url).replace(/#\{label\}/, label));
+			var $li = $(o.tabTemplate.replace(/#\{href\}/g, url).replace(/#\{label\}/g, label));
 			$li.data('destroy.tabs', true);
 
 			var id = url.indexOf('#') == 0 ? url.replace('#', '') : this.tabId( $('a:first-child', $li)[0] );
