Index: src/selector.js
===================================================================
--- src/selector.js	(revision 5795)
+++ src/selector.js	(working copy)
@@ -334,11 +334,9 @@
 				var tmp = [], type = m[3];
 
 				for ( var i = 0, rl = r.length; i < rl; i++ ) {
-					var a = r[i], z = a[ jQuery.props[m[2]] || m[2] ];
+					var a = r[i], 
+						z = jQuery.attr( a, m[2] ) || '';
 
-					if ( z == null || /style|href|src|selected/.test(m[2]) )
-						z = jQuery.attr(a,m[2]) || '';
-
 					if ( (type == "" && !!z ||
 						 type == "=" && z == m[5] ||
 						 type == "!=" && z != m[5] ||
