Index: src/ajax.js
===================================================================
--- src/ajax.js	(revision 3798)
+++ src/ajax.js	(working copy)
@@ -174,7 +174,7 @@
 		}
 
 		// Build temporary JSONP function
-		if ( s.dataType == "json" && (s.data && jsre.test( s.data ) || s.url.match(jsre)) ) {
+		if ( s.dataType == "json" && (s.data && s.data.match(jsre) || s.url.match(jsre)) ) {
 			jsonp = "jsonp" + jsc++;
 
 			// Replace the =? sequence both in the query string and the data
