--- jquery.validate.js	2008-05-30 13:51:52.000000000 -0600
+++ jquery.validate.js.patched	2008-05-30 13:51:07.000000000 -0600
@@ -35,7 +35,7 @@
 		if ( validator.settings.onsubmit ) {
 		
 			// allow suppresing validation by adding a cancel class to the submit button
-			this.find(".cancel:submit").click(function() {
+			this.find(".cancel:submit,.cancel:image").click(function() {
 				validator.cancelSubmit = true;
 			});
 		
