--- jquery-1.0.js.orig	2006-08-25 22:02:22.000000000 -0700
+++ jquery-1.0.js	2006-08-28 14:23:42.000000000 -0700
@@ -1253,6 +1253,21 @@
 		});
 	},
 
+    slideUpFade: function(speed, callback) {
+        return this.animate({height: "hide", opacity: "hide"}, speed, callback);
+    },
+
+    slideDownFade: function(speed, callback) {
+        return this.animate({height: "show", opacity: "show"}, speed, callback);
+    },
+
+    slideFadeToggle: function(speed, callback) {
+		return this.each(function() {
+			var state = $(this).is(":hidden") ? "show" : "hide";
+			$(this).animate({height: state, opacity: state}, speed, callback);
+		});
+    },
+
 	fadeIn: function(speed,callback){
 		return this.animate({opacity: "show"}, speed, callback);
 	},
