From e3d6c7316bd66f9c986be29eff99568171fa074d Mon Sep 17 00:00:00 2001
From: Yehuda Katz <wycats@gmail.com>
Date: Tue, 21 Jul 2009 00:48:11 -0700
Subject: [PATCH] Exposes contains

---
 libs/jquery.js |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/libs/jquery.js b/libs/jquery.js
index 5803cf5..5b021a3 100644
--- a/libs/jquery.js
+++ b/libs/jquery.js
@@ -62,4 +62,6 @@ jQuery.sibling = function(n, elem){
 	return r;
 };
 
+jQuery.contains = contains;
+
 return;
-- 
1.6.3.3

