diff --git a/src/css.js b/src/css.js
index 5d31757..073600e 100644
--- a/src/css.js
+++ b/src/css.js
@@ -214,7 +214,7 @@ jQuery.extend({
 if ( jQuery.expr && jQuery.expr.filters ) {
 	jQuery.expr.filters.hidden = function( elem ) {
 		var width = elem.offsetWidth, height = elem.offsetHeight,
-			skip = elem.nodeName.toLowerCase() === "tr";
+			skip = elem.nodeName.toLowerCase() === "tr" || elem.nodeName.toLowerCase() === "option" ;
 
 		return width === 0 && height === 0 && !skip ?
 			true :
