Index: jquery.fastTrigger.js
===================================================================
--- jquery.fastTrigger.js	(revision 5641)
+++ jquery.fastTrigger.js	(working copy)
@@ -57,7 +57,8 @@
 				for( var i in handlers ){
 					handler = handlers[i];
 					if(	any || handler.type == ns ){
-						e.data = handler.data;
+						e.data = handler.data;
+						e.handler = handler;
 						if( args )//call is slightly faster, thus preferred
 							handler.apply( this, args );
 						else
