Index: jquery.treeview.js
===================================================================
--- jquery.treeview.js	(revision 5363)
+++ jquery.treeview.js	(working copy)
@@ -218,9 +218,14 @@
 			}
 			
 			return this;
-		}
-	});
+		},
 	
+    expand: function() {
+      $(this).parents("li."+CLASSES.expandable).find(">div."+CLASSES.hitarea).click();
+    }
+  
+  });
+	
 	// classes used by the plugin
 	// need to be styled via external stylesheet, see first example
 	$.treeview = {};
