Index: core.js
===================================================================
--- core.js	(revision 3261)
+++ core.js	(working copy)
@@ -1093,7 +1093,7 @@
 	jQuery.fn[ n ] = function(h) {
 		return this[0] == window ?
 			jQuery.browser.safari && self["inner" + name] ||
-			jQuery.boxModel && Math.max(document.documentElement["client" + name], document.body["client" + name]) ||
+			jQuery.boxModel && Math.min(document.documentElement["client" + name], document.body["client" + name]) ||
 			document.body["client" + name] :
 		
 			this[0] == document ?
