Index: jquery/src/core.js
===================================================================
--- jquery/src/core.js	(revision 3477)
+++ jquery/src/core.js	(working copy)
@@ -923,6 +923,9 @@
 					!tags.indexOf("<col") &&
 					[ 2, "<table><tbody></tbody><colgroup>", "</colgroup></table>" ] ||
 
+					!tags.indexOf("<label") &&
+					[0, "", ""] ||
+
 					// IE can't serialize <link> and <script> tags normally
 					jQuery.browser.msie &&
 					[ 1, "div<div>", "</div>" ] ||
