Index: fx.magnifier.js
===================================================================
--- fx.magnifier.js	(revision 4508)
+++ fx.magnifier.js	(working copy)
@@ -34,10 +34,11 @@
 		o.magnification = o.magnification || 2;
 		o.baseline = o.baseline || 0;
 		o.verticalLine =  o.verticalLine != undefined ? o.verticalLine : -0.5;
+		o.elements = o.elements || '> *';
 		
 		this.pp = $(el).offset({ border: false });
 		
-		$('> *', el).each(function() {
+		$(o.elements, el).each(function() {
 			var co = $(this).offset({ border: false });
 			if(self.options.overlap) var cp = $(this).position();
 			self.items.push([this, co, [$(this).width(),$(this).height()], (cp || null)]);
