Index: javascripts/jquery.js
===================================================================
--- javascripts/jquery.js	(revision 189)
+++ javascripts/jquery.js	(working copy)
@@ -1179,6 +1179,11 @@
 	},
 
 	grep: function( elems, callback, inv ) {
+		// If a string is passed in for the function, make a function
+		// for it (a handy shortcut)
+		if ( typeof callback == "string" )
+			callback = eval("false||function(a,i){return " + callback + "}");
+
 		var ret = [];
 
 		// Go through the array, only saving the items
@@ -1191,6 +1196,11 @@
 	},
 
 	map: function( elems, callback ) {
+		// If a string is passed in for the function, make a function
+		// for it (a handy shortcut)
+		if ( typeof callback == "string" )
+			callback = eval("false||function(a,i){return " + callback + "}");
+
 		var ret = [];
 
 		// Go through the array, translating each of the items to their
