Index: src/core.js
===================================================================
--- src/core.js	(revision 4628)
+++ src/core.js	(working copy)
@@ -852,9 +852,9 @@
 		}
 		// Opera sometimes will give the wrong display answer, this fixes it, see #2037
 		if ( jQuery.browser.opera && name == "display" ) {
-			var save = elem.style.display;
-			elem.style.display = "block";
-			elem.style.display = save;
+			var tmp = elem.style.outline;
+			elem.style.outline = '0 solid black';
+			elem.style.outline = tmp;
 		}
 		
 		// Make sure we're using the right name for getting the float value
