Index: css.js
===================================================================
--- css.js	(revision 6617)
+++ css.js	(working copy)
@@ -110,7 +110,7 @@
 	},
 
 	curCSS: function( elem, name, force ) {
-		var ret, style = elem.style, filter;
+		var ret, style = elem.style, filter, defaultView = elem.ownerDocument.defaultView;
 
 		// IE uses filters for opacity
 		if ( !jQuery.support.opacity && name === "opacity" && elem.currentStyle ) {
