--- a/jquery/src/event.js
+++ b/jquery/src/event.js
@@ -283,7 +283,7 @@ jQuery.event = {
 		var originalEvent = event;
 		event = { originalEvent: originalEvent };
 		var props = "altKey attrChange attrName bubbles button cancelable charCode clientX clientY ctrlKey currentTarget data detail eventPhase fromElement handler keyCode metaKey newValue originalTarget pageX pageY prevValue relatedNode relatedTarget screenX screenY shiftKey srcElement target timeStamp toElement type view wheelDelta which".split(" ");
-		for ( var i=props.length; i; i-- )
+		for ( var i=props.length-1; i>=0; i-- )
 			event[ props[i] ] = originalEvent[ props[i] ];
 
 		// Mark it as fixed
