Index: event.js
===================================================================
--- event.js	(révision 6320)
+++ event.js	(copie de travail)
@@ -666,7 +666,7 @@
 
 jQuery.each( ("blur,focus,load,resize,scroll,unload,click,dblclick," +
 	"mousedown,mouseup,mousemove,mouseover,mouseout,mouseenter,mouseleave," +
-	"change,select,submit,keydown,keypress,keyup,error").split(","), function(i, name){
+	"change,select,submit,keydown,keypress,keyup,error,contextmenu").split(","), function(i, name){
 
 	// Handle event binding
 	jQuery.fn[name] = function(fn){
