Index: fx.js
===================================================================
--- fx.js	(revision 6173)
+++ fx.js	(working copy)
@@ -19,7 +19,7 @@
 
 jQuery.fn.extend({
 	show: function(speed,callback){
-		if ( speed ) {
+		if ( speed || speed === 0 ) {
 			return this.animate( genFx("show", 3), speed, callback);
 		} else {
 			for ( var i = 0, l = this.length; i < l; i++ ){
@@ -53,7 +53,7 @@
 	},
 
 	hide: function(speed,callback){
-		if ( speed ) {
+		if ( speed || speed === 0 ) {
 			return this.animate( genFx("hide", 3), speed, callback);
 		} else {
 			for ( var i = 0, l = this.length; i < l; i++ ){
