Index: jquery/src/css.js
===================================================================
--- jquery/src/css.js	(Revision 6533)
+++ jquery/src/css.js	(Working copy)
@@ -201,7 +201,7 @@
 
 				// Put in the new values to get a computed value out
 				elem.runtimeStyle.left = elem.currentStyle.left;
-				style.left = ret || 0;
+				style.left = (camelCase == "fontSize") ? "1em" : (ret || 0);
 				ret = style.pixelLeft + "px";
 
 				// Revert the changed values
