--- jquery.autocomplete.js
+++ (clipboard)
@@ -218,6 +218,11 @@
 		
 		var currentValue = $input.val();
 		
+		if (options.allowedChars != ''){
+			var reg = new RegExp('[^' + options.allowedChars + ']', 'g');
+			currentValue = currentValue.replace(reg, '');
+		}
+		
 		if ( !skipPrevCheck && currentValue == previousValue )
 			return;
 		
@@ -382,6 +398,8 @@
 	formatMatch: null,
 	autoFill: false,
 	width: 0,
+	allowedChars: '',
 	multiple: false,
 	multipleSeparator: ", ",
 	highlight: function(value, term) {