Index: support.js
===================================================================
--- support.js	(revision 6158)
+++ support.js	(working copy)
@@ -51,6 +51,10 @@
 		// (IE uses styleFloat instead of cssFloat)
 		cssFloat: !!a.style.cssFloat,
 
+		// Verify maxHeight property
+		// (IE 6 will be undefined)
+		maxHeight: div.style.maxHeight != undefined,
+
 		// Will be defined later
 		scriptEval: false,
 		noCloneEvent: true,
