--- jquery-1.2.2b2.js.orig	2008-01-02 02:50:04.000000000 -0800
+++ jquery-1.2.2b2.js	2008-01-02 02:56:14.000000000 -0800
@@ -836,7 +836,7 @@
 		if ( name.match( /float/i ) )
 			name = styleFloat;
 
-		if ( !force && elem.style[ name ] )
+		if ( !force && elem.style && elem.style[ name ] )
 			ret = elem.style[ name ];
 
 		else if ( document.defaultView && document.defaultView.getComputedStyle ) {
