Index: jquery.autocomplete.js
===================================================================
--- jquery.autocomplete.js	(revision 959)
+++ jquery.autocomplete.js	(revision 960)
@@ -507,7 +507,7 @@
 	multiple: false,
 	multipleSeparator: ", ",
 	highlight: function(value, term) {
-		return value.replace(new RegExp("(?![^&;]+;)(?!<[^<>]*)(" + term + ")(?![^<>]*>)(?![^&;]+;)", "gi"), "<strong>$1</strong>");
+		return value.replace(new RegExp("(?![^&;]+;)(?!<[^<>]*)(" + term.replace(new RegExp("([()[\\]\\\\{}?.+*|^$])","gi"),"\\$1") + ")(?![^<>]*>)(?![^&;]+;)", "gi"), "<strong>$1</strong>");
 	},
     scroll: true,
     scrollHeight: 180,
