Index: jquery.ui/ui.dialog.js
===================================================================
--- jquery.ui/ui.dialog.js 
+++ jquery.ui/ui.dialog.js 

@@ -178,5 +178,5 @@
 		
 		position: function(pos) {
-			var wnd = $(window), doc = $(document), minTop = top = doc.scrollTop(), left = doc.scrollLeft();
+			var wnd = $(window), doc = document.body, minTop = top = doc.scrollTop, left = doc.scrollLeft;
 			if ($.inArray(pos, ['center','top','right','bottom','left']) >= 0) {
 				pos = [pos == 'right' || pos == 'left' ? pos : 'center', pos == 'top' || pos == 'bottom' ? pos : 'middle'];
