Index: jquery/src/ajax.js
===================================================================
--- jquery/src/ajax.js	(revision 6274)
+++ jquery/src/ajax.js	(working copy)
@@ -250,7 +250,7 @@
 		if ( s.dataType == "script" && type == "GET" && parts
 			&& ( parts[1] && parts[1] != location.protocol || parts[2] != location.host )){
 
-			var head = document.getElementsByTagName("head")[0];
+			var head = document.getElementsByTagName("head")[0] || document.documentElement,
 			var script = document.createElement("script");
 			script.src = s.url;
 			if (s.scriptCharset)
