Index: ui/current/ui.mouse.js
===================================================================
--- ui/current/ui.mouse.js	(wersja 3817)
+++ ui/current/ui.mouse.js	(kopia robocza)
@@ -244,7 +244,7 @@
 		return ow;
 	}
 	function outerHeight(el) {
-		var $el = $(el), oh = $el.width();
+		var $el = $(el), oh = $el.height();
 		for (var i = 0, props = ['borderTopWidth', 'paddingTop', 'paddingBottom', 'borderBottomWidth']; i < props.length; i++)
 			oh += num($el, props[i]);
 		return oh;
