Index: jsparty/jquery/jquery.js
===================================================================
--- jsparty/jquery/jquery.js	(revision 62285)
+++ jsparty/jquery/jquery.js	(working copy)
@@ -2777,7 +2777,7 @@
 		function success(){
 			// If a local callback was specified, fire it and pass it the data
 			if ( s.success )
-				s.success( data, status );
+				s.success( data, status, xml );
 
 			// Fire the global callback
 			if ( s.global )
