Index: idrop.js
===================================================================
--- idrop.js	(revision 2642)
+++ idrop.js	(working copy)
@@ -145,9 +145,7 @@
 					
 					if (jQuery.iSort && iEL.dropCfg.s && jQuery.iDrag.dragged.dragCfg.so) {
 						iEL.dropCfg.el = jQuery('.' + iEL.dropCfg.a, iEL);
-						elm.style.display = 'none';
 						jQuery.iSort.measure(iEL);
-						elm.style.display = elm.dragCfg.oD;
 					}
 				}
 			}
