--- jquery-1.1.3a.js	2007-06-05 09:27:14.000000000 +0200
+++ jquery-1.1.3a_fix1143v3.js	2007-06-07 14:05:30.000000000 +0200
@@ -7,8 +7,8 @@ if(typeof window.jQuery == "undefined") 
  * Dual licensed under the MIT (MIT-LICENSE.txt)
  * and GPL (GPL-LICENSE.txt) licenses.
  *
- * $Date: 2007-05-21 00:49:26 +0000 (Mon, 21 May 2007) $
- * $Rev: 1938 $
+ * $Date: 2007-05-16 18:21:50 +0200 (Mi, 16 Mai 2007) $
+ * $Rev: 1910 $
  */
 
 // Global undefined variable
@@ -1678,8 +1678,8 @@ jQuery.extend({
 		var r = [], num = jQuery.mergeNum++;
 
 		for ( var i = 0, fl = first.length; i < fl; i++ )
-			if ( num != first[i].mergeNum ) {
-				first[i].mergeNum = num;
+			if ( !first[i].mergeNum || num != first[i].mergeNum[0] ) {
+				first[i].mergeNum = [num];
 				r.push(first[i]);
 			}
 
