Index: src/selector.js
===================================================================
--- src/selector.js	(revision 4628)
+++ src/selector.js	(working copy)
@@ -336,7 +336,7 @@
 				for ( var i = 0, rl = r.length; i < rl; i++ ) {
 					var a = r[i], z = a[ jQuery.props[m[2]] || m[2] ];
 					
-					if ( z == null || /href|src|selected/.test(m[2]) )
+					if ( z == null || /href|src|selected|align/.test(m[2]) )
 						z = jQuery.attr(a,m[2]) || '';
 
 					if ( (type == "" && !!z ||
